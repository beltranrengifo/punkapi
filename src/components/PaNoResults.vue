<template>
  <section class="pa-no-results text-center">
    <h2 class="mb-8">
      {{ text }}
    </h2>
    <p v-if="subtext">
      {{ subtext }}
    </p>
  </section>
</template>

<script>
export default {
  name: 'PaNoResults',

  props: {
    text: {
      type: String,
      default: 'Nothing found',
    },
    subtext: {
      type: String,
      default: '',
    },
  },
}
</script>
