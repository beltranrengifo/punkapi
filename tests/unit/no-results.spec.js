
import { shallowMount } from '@vue/test-utils'
import PaNoResults from '@/components/PaNoResults'

describe('PaNoResults component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(PaNoResults, {
      propsData: {
        text: 'No results',
        subtext: 'My subtext'
      },
    })
  })

  it('Component should find the title', () => {
    const title = wrapper.find('h2')
    expect(title.is('h2')).toBe(true)
  })

  it('Component should render the text prop', () => {
    const title = wrapper.find('h2')
    expect(title.text()).toContain('No results')
  })

  it('Component should render the subtext prop', () => {
    const p = wrapper.find('p')
    expect(p.text()).toContain('My subtext')
  })

  it('Component should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
