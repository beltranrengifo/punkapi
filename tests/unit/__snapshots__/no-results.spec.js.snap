// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaNoResults component Component should match the snapshot 1`] = `
<section
  class="pa-no-results text-center"
>
  <h2
    class="mb-8"
  >
    
    No results
  
  </h2>
   
  <p>
    
    My subtext
  
  </p>
</section>
`;
